ATB/DataAccess/Repositories/Implementations/CsvBookingRepository.cs

```diff
@@ -0,0 +1,85 @@
+using DataAccess.CsvHelperService;
+using DataAccess.Enums;
+using DataAccess.Models;
+using DataAccess.Repositories.Interfaces;
+using DataAccess.SearchCriteria;
+
+namespace DataAccess.Repositories.Implementations;
+
+public class CsvBookingRepository : IBookingRepository
+{
+    private readonly string _pathToCsv;
+    private readonly ICsvFileService<Booking> _csvFileService;
+    private List<Booking> _bookingsCache;
+    private bool _isCacheInitialized = false;
+
+    public CsvBookingRepository(ICsvFileService<Booking> csvFileService, string pathToCsv)
+    {
+        _csvFileService = csvFileService;
+        _pathToCsv = pathToCsv;
+    }
+
+    private async Task InitializeCacheAsync()
+    {
+        if (!_isCacheInitialized)
+        {
+            _bookingsCache = await _csvFileService.ReadFromCsvAsync(_pathToCsv);
+            _isCacheInitialized = true;
```

Comment:
I believe you are trying to mimic a cache behavior but let's think together about the current implementation, 
* Currently you are initializing the cache, and what does the initialization do is loading all objects in memory which in real world would be similar to loading all table's records in memory , is that practical? should not we cache only the requested entities?
* The general flow of caching is as following
    *  In case of retrieving/fetching data, we check if it does exit in the cache, if so we return it, if not we retrieve it from db/csv, we cache it and return result to user, so next time it will be cached and we won't access db/csv unless the entry got invalidated.
    * In case of Deleting/Updating we invalidate the cache, no need to cache anything that got newly created.

Think about the mentioned flow, it should cover all cases.

Note: this comment is not a blocker, it's up to you if you want to address it.